### PR TITLE
Task/improve documentatio

### LINKED ISSUE #426: Documentation typos and aspects to fix
While reading the documentation I found the following typos which should be fixed (I include the section and the error):

1. Section https://github.com/telefonicaid/iotagent-node-lib#transactions, formar -> former
2. Section https://github.com/telefonicaid/iotagent-node-lib#description-11, creatinos -> creations
3. The iotagentLib.listDevices() is documented twice: 1) https://github.com/telefonicaid/iotagent-node-lib/tree/develop#iotagentliblistdevices and 2) https://github.com/telefonicaid/iotagent-node-lib/tree/develop#iotagentliblistdevices-1 (in my opinion the second one is better :))
4. Section https://github.com/telefonicaid/iotagent-node-lib/tree/develop#iotagentlibunsuscribe, iotagentLib.suscribe -> iotagentLib.subscribe
5. The options detailed as included in the Agent Console are outdated (some options are missing): https://github.com/telefonicaid/iotagent-node-lib/tree/develop#agent-console
6. The options detailed as included in the Agent Tester are outdated (some options are missing): https://github.com/telefonicaid/iotagent-node-lib/tree/develop#command-line-testing
7. Section https://github.com/telefonicaid/iotagent-node-lib/tree/develop#device-model, the provided example of value for the `protocol` refers to the timezone and not to a protocol.
8. Section https://github.com/telefonicaid/iotagent-node-lib/tree/develop#-configuration-api, includes a reference to https://github.com/telefonicaid/iotagent-node-lib/issues/33 which seems to be already solved.
9. Section https://github.com/telefonicaid/iotagent-node-lib/tree/develop#overview-1, allo -> all

### LINKED ISSUE #427: Add the names of the DB fields to the documentation
The DB colections closely resemble the API resources, but the actual name of the fields change in multiple cases. Those cases should be documented in the model.

### PR DESCRIPTION
**Pull request by dmoranj**
Fixes #426
FIxes #427

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1254,13 +1269,19 @@ Payload example:
...
+For some services, there will be no need to provision individual devices, but it will make more sense to provision
+different device groups, each of one mapped to a different type of entity in the context broker. How the type of entity
+is assigned to a device will depend on the Southbound technology (e.g.: path, port, APIKey...). Once the device has an
+assigned type, its configuration values can be extracted from those of the type.
 
 The IoT Agents provide two means to define those device groups:
 * Static **Type Configuration**: configuring the `ngsi.types` property in the `config.js` file.
-* Dinamic **Configuration API**: making use of the API URLS in the configuration URI, `/iot/services`. Please, note that the configuration API manage servers under an URL that requires the `server.name` parameter to be set (the name of the IoT Agent we are using). If no name is configured `default` is taken as the default one.
+* Dinamic **Configuration API**: making use of the API URLS in the configuration URI, `/iot/services`. Please, note
```
Comment (gtorodelvalle) [inline README.md]: Dinamic -> Dynamic

**Comment by gtorodelvalle** [comment]
LGTM ;)

File: README.md [hunk 2]
```diff
@@ -1038,22 +1049,26 @@ same headers, as received by the Device Provisioning API (this two headers are,
...
-| attributes          | List of active attributes of the device	| `[ { "name": "attr_name", "type": "string" } ]` |
-| lazy                | List of lazy attributes of the device	| `[ { "name": "attr_name", "type": "string" } ]` |
-| commands            | List of commands of the device	| `[ { "name": "attr_name", "type": "string" } ]` |
-| internal_attributes | List of internal attributes with free format for specific IoT Agent configuration | LWM2M mappings from object URIs to attributes |
-| static_attributes   | List of static attributes to append to the entity. All the updateContext requests to the CB will have this set of attributes appended.	| `[ { "name": "attr_name", "type": "string" } ]` |
+
+The next table shows the information held in the Device resource. The table also contains the correspondence between
+the API resource fields and the same fields in the database model.
+
+| Attribute | DB Field | Definition                                     | Example of value                      |
```
Comment (fgalan) [inline README.md]: I'd suggest to use "Payload field" (or some alternative not using "attribute") to avoid confussion.

File: README.md [hunk 3]
```diff
@@ -1269,20 +1290,39 @@ The following sections show the available operations for the Configuration API.
...
+* **type**:
+* **trust**:
+* **cbHost**:
+* **lazy**:
+* **commands**:
+* **active**:
+* **staticAttributes**:
+* **internalAttributes**:
+
+| Attribute | DB Field | Definition                                     |
```
Comment (fgalan) [inline README.md]: I'd suggest to use "Payload field" (or some alternative not using "attribute") to avoid confussion.

**Pull request merged**